agent/exec/dockerapi/container.go

```diff
@@ -137,13 +138,26 @@ func (c *containerConfig) exposedPorts() map[nat.Port]struct{} {
 	return exposedPorts
 }
 
+func (c *containerConfig) getGenericResourcesEnv() []string {
+	i := 0
+	env := make([]string, len(c.task.AssignedGenericResources))
+	for k, v := range c.task.AssignedGenericResources {
+		env[i] = genericresource.GetEnvFormat(k, v)
+		i++
+	}
```

Comment:
This is another place that could use the:

```go
env := make([]string, 0, len(c.task.AssignedGenericResources))
for k, v := range c.task.AssignedGenericResources {
	env = append(env, genericresource.GetEnvFormat(k, v))
}
```

approach, but it doesn't really matter, and the existing code is also fine.